packages/react-intl/src/formatters/message.ts

```diff
@@ -20,6 +21,13 @@ import IntlMessageFormat, {
 } from 'intl-messageformat';
 import {MissingTranslationError, MessageFormatError} from '../error';
 
+type FormatMessageValues = Record<
+  string,
+  | PrimitiveType
+  | React.ReactNode
+  | FormatXMLElementFn<React.ReactNode, React.ReactNode>
+>;
+
```

Comment:
I created a custom type for `FormatMessageValues`, I tend to move it into `./types.ts` and reference it in all places, where `values` are used. Is there any notable differences in `values` between Component API and imperative API or could I use the exact same type everywhere?

Reply:
yeah this will mess up functional component

Reply:
Custom type for `FormatMessageValues` was discarded..